rholang/src/test/scala/coop/rchain/rholang/interpreter/ReduceSpec.scala

```diff
@@ -878,7 +934,10 @@ class ReduceSpec extends FlatSpec with Matchers with PersistentStoreTester {
 
   "variable references" should "be substituted before being used." in {
     implicit val errorLog = new Runtime.ErrorLog()
-
+    val splitRandResult = rand.splitByte(3)
+    val splitRandSrc = rand.splitByte(3)
+    splitRandResult.next()
```

Comment:
I didn't do good job reviewing your previous PR that added `Blake2b512Random` but this is not functional. I understand that mutation is sometimes more performant.

Reply:
The functional version of next() would require array copies.
In actual use, we want the overwrite anyway, because we **don't** want to get the same random numbers twice.

It's only for testing that we intentionally break the API requirements so that we can make sure that we're repeatable.

Reply:
Make sense :thumbsup: